### DISCUSSION TITLE
Unity SDK 2.0 Alpha Feedback

### DISCUSSION THREAD

**Discussion by bfercher** [readyplayerme/rpm-unity-sdk-core#35] [Q&A]
Hey Developers,
we appreciate all the feedback you have for us on the alpha version of the new Unity SDK by answering the following questions:
- How was your experience upgrading/adapting/installing Unity SDK 2.0?
- Did you encounter any issues?
- How can we improve the installation/upgrading experience?

Cheers,
Bernhard

**Comment by Morbeavus** [comment]
Hello

> How can we improve the installation/upgrading experience?

please add the packages to openUPM https://openupm.com/

Best,
Morb

**Reply by srcnalt** [reply]
Hi, @Morbeavus thank you for your feedback.

We decided not to start with OpenUPM since it requires more steps to bring into the project compared to Git URLs. However, we are considering the possibility and most likely will add the packages there.

**Comment by davidyrgilbert** [comment]
Hey! Had this on my todo list for a while, thank you a lot for putting everything on github, it is very awesome! Our transition wasn't as smooth as it could have been due to us having manually implemented GLTFast loading and modified the assembly definitions, but in the end everything worked out well. Due to the compile errors there, the dependency modules weren't installed at first, but after fixing those it worked. 

Personally not quite sure if I'm a fan of handling dependency installation in a custom script, but it worked fine in the end. 

Being on github is really great, and your response time on our Pull Request a bit ago was also top, so this is a great change!

**Reply by srcnalt** [reply]
Thank you for your feedback @davidyrgilbert, we can update our [migration guide](https://wolf3d.notion.site/Unity-SDK-2-0-Migration-Guide-32ca364853f44b35ac527bf61fa100ed) considering these points. 

Unity Package Manager still does not support Git URLs as dependencies so we had to go for this approach to begin with. One way to update things might be to look closely into UPM as suggested in the comment above. It was not our first option since many Unity users may not be using it and it requires Node installation. 

Really happy to shape the SDK with your suggestions.

**Comment by JanFromNaomiAnimation** [comment]
Hi guys, is there some getter for cache directory path? 

In SDK 1.0 avatars were stored somewhere in <Persistent Data Path>/Avatars/<avatar GUID>/
In SDK 2.0 path is <Persistent Data Path>/Ready Player Me/Avatars/<avatar GUID>/

Maybe its already there somewhere, maybe I just didnt find it. But it would be great to have some getter for that. E.g. in AvatarCache. Our integrations may become much more issues-bulletproof against those changes. I fully understand, this path will not be changing often, but it can be nice to have it and not think about it anymore (in case, when we are using your cache for something)

Thnx for your amazing work on this service!

**Reply by srcnalt** [reply]
Hi @JanFromNaomiAnimation, only thing changed in the path is addition of `Ready Player Me`
We figured some projects might already have a `Avatars` folder in the root and that creates confusion, so we wanted to make sure everything that is generated via RPM goes under `Assets/Ready Player Me` folder. Including setting scriptable objects.

In the version 2.0, 
- For avatars loaded through `Avatar Loader Window` the cache is `Ready Player Me/Avatars/<avatar_guid>`
- For avatars loaded via script the cache is `C:\Users\<user>\AppData\LocalLow\<company_name>\<project_name>\Ready Player Me/Avatars/<avatar_guid>`

**Comment by JanFromNaomiAnimation** [comment]
Is there any official release date for SKD 2.0?
thnx

**Reply by srcnalt** [reply]
Finalizing the docs and doing QA atm, we are aiming for next week. 🤞

**Comment by CadencePr** [comment]
I'm getting this error message and couldn't make it work. (Unity 2022.2.9, gltFast 4.8.3)
Library\PackageCache\com.readyplayerme.avatarloader@64b065ec8f\Runtime\Operations\GltFastGameObjectInstantiator.cs(15,13): error CS0246: The type or namespace name '**InstantiationSettings**' could not be found (are you missing a using directive or an assembly reference?)

**Reply by srcnalt** [reply]
Hi @CadencePr, SDK requires glTFast 5.0.0, can you try updating from core module to see if the issue is fixed?

**Reply by CadencePr** [reply]
@srcnalt Somehow, the 'Update' button in the package manager said it is up to date. I manually added glTFast.git#v5.0.0 URL to update the package and it resolved the issue. Thank you!
